cpp/src/column/column_factories.cpp

```diff
@@ -81,4 +85,50 @@ std::unique_ptr<column> make_fixed_width_column(
   return make_numeric_column(type, size, state, stream, mr);  
 }
 
+struct column_from_scalar_dispatch 
+{
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+  operator()( scalar const& value, size_type size,
+              rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    auto output_column = make_fixed_width_column(value.type(),size,mask_state::UNALLOCATED,stream,mr);
+    auto view = output_column->mutable_view();
+    experimental::detail::fill_in_place(view,0,size,value,stream);
+    return output_column;
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_same<cudf::string_view, T>::value, std::unique_ptr<cudf::column>>
+  operator()( scalar const& value, size_type size,
+              rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    auto p_scalar = static_cast<experimental::scalar_type_t<T> const*>(&value);
```

Comment:
You can cast references. 
```suggestion
    auto p_scalar = static_cast<experimental::scalar_type_t<T> const&>(value);
```